pkg/apis/extensions/validation/validation.go

```diff
@@ -332,14 +332,27 @@ func ValidateDeploymentStatus(status *extensions.DeploymentStatus, fldPath *fiel
 	return allErrs
 }
 
+func ValidateSelector(update *metav1.LabelSelector, old *metav1.LabelSelector, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if update == nil || old == nil {
+		return allErrs
+	}
+	if !reflect.DeepEqual(update, old) {
+		allErrs = append(allErrs, field.Required(fldPath, "Spec.Selector must not be changed prior to any update operation"))
+	}
+	return allErrs
+}
+
 func ValidateDeploymentUpdate(update, old *extensions.Deployment) field.ErrorList {
 	allErrs := apivalidation.ValidateObjectMetaUpdate(&update.ObjectMeta, &old.ObjectMeta, field.NewPath("metadata"))
+	allErrs = append(allErrs, ValidateSelector(update.Spec.Selector, old.Spec.Selector, field.NewPath("spec").Child("selector"))...)
 	allErrs = append(allErrs, ValidateDeploymentSpec(&update.Spec, field.NewPath("spec"))...)
 	return allErrs
 }
 
 func ValidateDeploymentStatusUpdate(update, old *extensions.Deployment) field.ErrorList {
 	allErrs := apivalidation.ValidateObjectMetaUpdate(&update.ObjectMeta, &old.ObjectMeta, field.NewPath("metadata"))
+	allErrs = append(allErrs, ValidateSelector(update.Spec.Selector, old.Spec.Selector, field.NewPath("spec").Child("selector"))...)
```

Comment:
Referring [this](https://github.com/kubernetes/kubernetes/blob/6a644c25f8a228d16826bb16b0a6e6779c2e3bfe/pkg/registry/extensions/deployment/strategy.go#L104-L110), any changes to Spec are ignored. Hence, we shouldn't need to do anything regarding `Spec.Selector` for `Validate*StatusUpdate()` functions.